### PR TITLE
Fix for #11710

### LINKED ISSUE #11710: Entire Mautic Installation 500 Internal Server Error After Clicking Webview Link in Email
### Mautic Version

4.4.x series

### PHP version

8.0.25

### What browsers are you seeing the problem on?

Chrome

### What happened?

Similar to the Issue [11650](https://github.com/mautic/mautic/issues/11650) I have a Server Error 500 after testing a web view from an Email Newsletter I created. 

I used an incognito tab to subscribe to the newsletter and opened the DOI email to check it out. I used the web view link to test this as well and afterwards my Mautic Dashboard ended in a 500 error. The log messages reads: 

 `[2022-11-13 17:11:38] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "method_exists(): Argument #1 ($object_or_class) must be of type object|string, null given" at /var/www/virtual/USERNAME_REMOVED/mautic.moorwald.com/app/bundles/CoreBundle/EventListener/DashboardSubscriber.php line 101 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): method_exists(): Argument #1 ($object_or_class) must be of type object|string, null given at /var/www/virtual/croci/mautic.moorwald.com/app/bundles/CoreBundle/EventListener/DashboardSubscriber.php:101)"} {"hostname":"HOSTNAME REMOVED","pid":28948}`
 
 

### How can we reproduce this issue?

Step 1: 

### Relevant log output

```shell
[2022-11-13 17:11:38] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "method_exists(): Argument #1 ($object_or_class) must be of type object|string, null given" at /var/www/virtual/USERNAME_REMOVED/mautic.moorwald.com/app/bundles/CoreBundle/EventListener/DashboardSubscriber.php line 101 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): method_exists(): Argument #1 ($object_or_class) must be of type object|string, null given at /var/www/virtual/croci/mautic.moorwald.com/app/bundles/CoreBundle/EventListener/DashboardSubscriber.php:101)"} {"hostname":"HOSTNAME REMOVED","pid":28948}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://app.bountysource.com/issues/121828593-entire-mautic-installation-500-internal-server-error-after-clicking-webview-link-in-email?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by TonyBogdanov**
Solution taken from: https://github.com/mautic/mautic/issues/11710#issuecomment-1354301693

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Yes
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #11710

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

### CODE REVIEW

File: app/bundles/CoreBundle/EventListener/DashboardSubscriber.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ public function onWidgetDetailGenerate(WidgetDetailEvent $event)
                     if (!empty($log['bundle']) && !empty($log['object']) && !empty($log['objectId'])) {
                         try {
                             $model = $this->modelFactory->getModel($log['bundle'].'.'.$log['object']);
-                            $item  = $model->getEntity($log['objectId']);
+                            $item  = $model->getEntity($log['objectId']) ?: '';
                             if (method_exists($item, $model->getNameGetter())) {
```
Comment (escopecz) [inline app/bundles/CoreBundle/EventListener/DashboardSubscriber.php]: I'd rather suggest this:
```suggestion
                            $item  = $model->getEntity($log['objectId']);
                            if ($item && method_exists($item, $model->getNameGetter())) {
```
The first argument to [method_exists](https://www.php.net/manual/en/function.method-exists.php) should be an object or a class. An empty string might be deprecated in the future PHP versions too.

**Review by escopecz — APPROVED** [review]
This change makes sense and won't hurt. Thanks for taking care of this! 👍

**Comment by escopecz** [comment]
Please check the failed test.

**Comment by TonyBogdanov** [comment]
> Please check the failed test.

Tried setting up Mautic locally following the guide, but `ddev start` gives me an error:
```
container 'db' exited, please use 'ddev logs -s db' to find out why it failed
>> + cp -r /home/.my.cnf /home/.bashrc /nonexistent/
>> cp: target '/nonexistent/' is not a directory
```

The failed test doesn't seem (to me at least) to have anything to do with the PR code and I simply don't have the time to familiarize myself with `ddev` and figure out why it won't start. Sorry.

**Comment by escopecz** [comment]
Aha, I see. I missed the target branch. The `4.x` branch is dead now. Please rebase to `5.x`. If you want to make a backport to Mautic 4, create a duplicate PR based on the `4.4` branch. There must be both if you want this to be fixed also for Mautic 4.

**Comment by TonyBogdanov** [comment]
@escopecz Just pulled `4.4` and realized the code was changed in https://github.com/mautic/mautic/commit/d154c6c598ccf039b18439e78d6c5ff45aaf3a55, which is part of `4.4.8` (maybe also `4.4.7`) and now I realize I haven't seen the issue after upgrading to `4.4.8`.

This PR is therefore unnecessary, so I'll be closing it. Sorry for the back-and-forth.

**Pull request closed without merging**